### PR TITLE
Update output for Data Preparation processing

### PR DESCRIPTION
**Pull request by fernst**
Updated data preparation proto to only carry binary encoded data preparation contents after processing the data preparation definition.

### CODE REVIEW

File: core/actions/data_preparation.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class DataPreparation extends ActionBuilder<dataform.DataPreparation> {
     config.filename = resolveActionsConfigFilename(config.filename, configPath);
     const dataPreparationContents = nativeRequire(config.filename).asJson;
     const dataPreparationDefinition = parseDataPreparationDefinitionJson(dataPreparationContents);
```
Comment (Ekrekr) [inline core/actions/data_preparation.ts:38]: Nit: the term `dataPreparationContents` is overloaded now, so should be renamed - something like `dataPreparationAsJson = ...`

File: packages/@dataform/core/BUILD [hunk 2]
```diff
@@ -99,7 +93,6 @@ pkg_npm_tar(
         ":bundle_with_license",
         ":configs_proto_with_license",
         ":core_proto_with_license",
-        ":data_preparation_proto_with_license",
```
Comment (Ekrekr) [inline packages/@dataform/core/BUILD:102]: You're still going to need this exported, for the decode step

**Review by Ekrekr — CHANGES_REQUESTED** [review]

File: core/actions/data_preparation.ts (hunk 1 above)
Comment (fernst) [reply core/actions/data_preparation.ts]: Updated

File: packages/@dataform/core/BUILD (hunk 2 above)
Comment (fernst) [reply packages/@dataform/core/BUILD]: I added this back as discussed.

File: protos/core.proto [hunk 3]
```diff
@@ -268,7 +267,10 @@ message DataPreparation {
 
   bool disabled = 6;
 
-  DataPreparationDefinition data_preparation = 7;
+  reserved 7;
```
[the remark this answers is not in the document]
Comment (fernst) [reply protos/core.proto]: Updated

File: core/main_test.ts [hunk 4]
```diff
@@ -945,62 +945,7 @@ nodes:
...
-                            }
-                          ]
-                        }
-                      }
-                    }
-                  }
-                }
-              ]
-            }
+            dataPreparationContents: "Cn4KBW5vZGUxEhASDgoDcHJqEgJkcxoDc3JjIlAiFgoUCgFhEgVJTlQ2NBoITlVMTEFCTEUqGyIZEhcKFQoBYRIGU1RSSU5HGghOVUxMQUJMRTIZChcKFQoBYRIGU1RSSU5HGghOVUxMQUJMRSoRCg8KA3ByahICZHMaBGRlc3Q=",
```
Comment (chtyim) [inline core/main_test.ts:948]: This is probably generated by some external script? Might be good to include that script / instruction on how to update this string.

Comment (Ekrekr) [reply core/main_test.ts:948]: Good shout - @fernst if you bring back what you put here before, and just call `dataform.DataPreparationDefinition.encode(` on it in the test part, then its generation becomes clear

**Comment by Ekrekr** [comment]
Our tests are being flakey atm because of the `Invalid JWT Signature` error - if you rerun them they should go away

image

File: core/main_test.ts (hunk 4 above)
Comment (fernst) [reply core/main_test.ts:948]: Good point. I'll add this logic so it's obvious.

**Pull request merged**